### PR TITLE
Initial import of package utility

### PR DESCRIPTION
**Pull request by benoitf**
packaging
command provided is "theia:plugin <package>"

### CODE REVIEW

**Review by evidolob — APPROVED** [review]

File: src/validate.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+        // TODO: check for yarn.lock
+
+        return Promise.resolve(true);
+    }
+
+    public validateEngine(pluginPackageJson: any, packageJsonPath: string): void {
+
+        const missingEngineSuggest = "You need to add "
+            + chalk.italic('\n  "engines": \{\n    "theia": "latest"\n  \}\n')
```
Comment (akurinnoy) [inline src/validate.ts]: I suppose here should be `"`**`theiaPlugin`**`": "latest"` instead of `"`**`theia`**`": "latest"`

Comment (benoitf) [reply src/validate.ts]: I was thinking it was the name of the product (theia) but if you think it's better to say it's the name of the component I'm fine as well.

Comment (akurinnoy) [reply src/validate.ts]: I think engine name should be related to the theia's plugin system but not to theia itself. 
WDYT?

Comment (benoitf) [reply src/validate.ts]: I will update the code, maybe there will other changes before it goes public when other will review, but for now let's make it to theia plugin's system version

Comment (benoitf) [reply src/validate.ts]: fixed

File: tests/validate/missing-engine-theia-package.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+    "engines": {
+        
+    }
+}
```
Comment (akurinnoy) [inline tests/validate/missing-engine-theia-package.json]: .json files in tests/validate are missing EOL

**Review by akurinnoy — APPROVED** [review]

File: tests/validate/missing-engine-theia-package.json (hunk 2 above)
Comment (benoitf) [reply tests/validate/missing-engine-theia-package.json]: yes I will fix these EOL

**Comment by benoitf** [comment]
ok, fixed

**Pull request merged**